### PR TITLE
Refactor:Remove the dependency of the Psi in the FFT module

### PR DESCRIPTION
**Pull request by A-006**
### Unit Tests and/or Case Tests for my changes
- In the module_pw/test,there is a Makefile which test the module_fft.Modify the file to test the memory safety and correctness of the fft.

### What's changed?
- Remove the dependency of the psi.h,move the check func of the cufft into the module_fft.

### CODE REVIEW

File: source/module_basis/module_pw/module_fft/kernel/fft_cuda_func.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        return "CUFFT_NOT_IMPLEMENTED";
+    case CUFFT_LICENSE_ERROR:
+        return "CUFFT_LICENSE_ERROR";
+    case CUFFT_NOT_SUPPORTED:
+        return "CUFFT_NOT_SUPPORTED";
+    }
+    return "<unknown>";
+}
+
+#define CHECK_CUFFT(func)                                                                                              \
```
Comment (mohanchen) [inline source/module_basis/module_pw/module_fft/kernel/fft_cuda_func.h]: why there are several '\' exist in the code?

Comment (A-006) [reply source/module_basis/module_pw/module_fft/kernel/fft_cuda_func.h]: The code in this section is only called in `module_fft`. Would splitting `check_func` make it easier to maintain?

Comment (mohanchen) [reply source/module_basis/module_pw/module_fft/kernel/fft_cuda_func.h]: It doesn't make any sense.

Comment (A-006) [reply source/module_basis/module_pw/module_fft/kernel/fft_cuda_func.h]: Okay, this part has been reverted.

**Review by Qianruipku — APPROVED** [review]

**Review by mohanchen — APPROVED** [review]

**Pull request merged**